lib/Parser/ptree.h

```diff
@@ -615,6 +618,23 @@ struct PnFinally : PnStmt
     ParseNodePtr pnodeBody;
 };
 
+struct PnSpecialName : PnPid
+{
+    bool isThis : 1;
+    bool isSuper : 1;
```

Comment:
enum (nit)

Reply:
Agree but I'm going to punt that until we add another special name. Unless you feel strongly. 😃